test/Microsoft.AspNetCore.ResponseCompression.Tests/ResponseCompressionMiddlewareTest.cs

```diff
@@ -793,13 +793,24 @@ private void CheckResponseCompressed(HttpResponseMessage response, int expectedB
         {
             IEnumerable<string> contentMD5 = null;
 
+            var containsVaryAcceptEncoding = false;
+            foreach (var value in response.Headers.GetValues(HeaderNames.Vary))
+            {
+                if (value.Contains(HeaderNames.AcceptEncoding))
```

Comment:
In this case either would work since the tests only set one value to the Vary header but in general when we are dealing with headers, we can't just assume each string in StringValues contains one value. We ran into this problem when we received Header["Cache-Control"] = new[] {"no-cache,no-store"} from MVC for example. So it's best to check using contains rather than exact match. 

In the same regard, we also shouldn't be assuming the format of the headers. For example, the test could break with a bad Vary header like Headers["Vary"] = new [] {"badValue=Accept-Encoding"} but I ignored this corner case in the tests.